src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -84,20 +87,25 @@ public object VisitSqlRoot(SqlRootExpression expression, SearchOptions context)
                     .Append(V1.Resource.Version, resourceTableAlias).Append(", ")
                     .Append(V1.Resource.IsDeleted, resourceTableAlias).Append(", ")
                     .Append(V1.Resource.ResourceSurrogateId, resourceTableAlias).Append(", ")
-                    .Append(V1.Resource.RequestMethod, resourceTableAlias).Append(", ")
+                    .Append(V1.Resource.RequestMethod, resourceTableAlias).Append(", ");
+
+                // If there's a table expression, use the previously selected bit, otherwise everything in the select is considered a match
+                StringBuilder.Append(expression.TableExpressions.Count > 0 ? "cast(isMatch as bit) as isMatch, " : "cast(1 as bit) as isMatch, ");
+
+                StringBuilder
                     .AppendLine(V1.Resource.RawResource, resourceTableAlias);
             }
 
-            StringBuilder.Append("FROM ").Append(V1.Resource).AppendLine(" r");
+            StringBuilder.Append("FROM ").Append(V1.Resource).Append(" ").AppendLine(resourceTableAlias);
```

Comment:
Was this deliberate? In other cases, the aliasing occurs right after the expression